pandas/io/parsers/base_parser.py

```diff
@@ -1094,13 +1096,27 @@ def _make_date_converter(
     date_parser=None,
     dayfirst: bool = False,
     cache_dates: bool = True,
+    date_format: str | None = None,
 ):
+    if date_parser is not None:
```

Comment:
just for date_parser, or date_format too?

Reply:
My concern is about backwards compatibility. If they pass None for date_format it fails immediately. Otherwise we could break existing code.

whatever you prefer